test/integration/dialects/mysql/connector-manager.test.js

```diff
@@ -87,8 +87,11 @@ if (dialect === 'mysql') {
         .then(connection => {
           // Save current connection
           conn = connection;
-          // simulate a unexpected end
-          connection.close();
+          // simulate a unexpected end from MySQL2
+          connection._protocolError = new Error('Connection lost: The server closed the connection.');
+          connection._protocolError.fatal = true;
+          connection._protocolError.code = 'PROTOCOL_CONNECTION_LOST';
+          connection._notifyError(connection._protocolError);
```

Comment:
The error event handler no longer destroys the connection resource and instead just noop/debug.  If we don't have a listener on the error event then we will have an uncaught exception whenever the connection emits an error.  Instead we check the validity of a connection before acquisition by setting the `testOnBorrow` property to true.